examples/deployment_crud.py

```diff
@@ -13,96 +13,137 @@
+            resp.metadata.name,
+            resp.metadata.generation,
+            resp.spec.template.spec.containers[0].image,
+        )
+    )
 
-def update_deployment(api_instance, deployment):
     # Update container image
     deployment.spec.template.spec.containers[0].image = "nginx:1.16.0"
-    # Update the deployment
-    api_response = api_instance.patch_namespaced_deployment(
-        name=DEPLOYMENT_NAME,
-        namespace="default",
-        body=deployment)
-    print("Deployment updated. status='%s'" % str(api_response.status))
-
-
-def delete_deployment(api_instance):
-    # Delete deployment
-    api_response = api_instance.delete_namespaced_deployment(
-        name=DEPLOYMENT_NAME,
-        namespace="default",
-        body=client.V1DeleteOptions(
-            propagation_policy='Foreground',
-            grace_period_seconds=5))
-    print("Deployment deleted. status='%s'" % str(api_response.status))
 
+    # patch the deployment
+    resp = api.patch_namespaced_deployment(
+        name=DEPLOYMENT_NAME, namespace="default", body=deployment
+    )
 
-def main():
-    # Configs can be set in Configuration class directly or using helper
-    # utility. If no argument provided, the config will be loaded from
-    # default location.
-    config.load_kube_config()
-    apps_v1 = client.AppsV1Api()
+    print("\n[INFO] deployment's container image updated.\n")
+    print("%s\t%s\t\t\t%s\t%s" % ("NAMESPACE", "NAME", "REVISION", "IMAGE"))
+    print(
+        "%s\t\t%s\t%s\t\t%s\n"
+        % (
+            resp.metadata.namespace,
+            resp.metadata.name,
+            resp.metadata.generation,
+            resp.spec.template.spec.containers[0].image,
+        )
+    )
 
-    # Uncomment the following lines to enable debug logging
-    # c = client.Configuration()
-    # c.debug = True
-    # apps_v1 = client.AppsV1Api(api_client=client.ApiClient(configuration=c))
 
-    # Create a deployment object with client-python API. The deployment we
-    # created is same as the `nginx-deployment.yaml` in the /examples folder.
-    deployment = create_deployment_object()
+    # update `spec.template.metadata` section
+    # to add `kubectl.kubernetes.io/restartedAt` annotation
+    deployment.spec.template.metadata.annotations = {
+        "kubectl.kubernetes.io/restartedAt": datetime.datetime.utcnow()
+        .replace(tzinfo=pytz.UTC)
+        .isoformat()
+    }
 
-    create_deployment(apps_v1, deployment)
+    # patch the deployment
+    resp = api.patch_namespaced_deployment(
+        name=DEPLOYMENT_NAME, namespace="default", body=deployment
+    )
 
-    update_deployment(apps_v1, deployment)
+    print("\n[INFO] deployment `nginx-deployment` restarted.\n")
+    print("%s\t\t\t%s\t%s" % ("NAME", "REVISION", "RESTARTED-AT"))
+    print(
+        "%s\t%s\t\t%s\n"
+        % (
+            resp.metadata.name,
+            resp.metadata.generation,
+            resp.spec.template.metadata.annotations,
+        )
+    )  
```

Comment:
to fix the pycodestyle test failure, please remove the trailing spaces in this line.

Reply:
@roycaihw, I fixed the whitespaces issues but the job still remain failing.

Another thing to note:

Just to figure out what exactly amongst the new changes I added,  is a breaking change. I started by adding just the following  import commands (on top of the example file in master branch)

```
+ import datetime
+ import pytz
   from kubernetes import client, config
```

The job is still failing. https://travis-ci.org/github/kubernetes-client/python/jobs/770894221

I'm not sure if I'm missing something really simple here?

Reply:
pycodestyle is asking for some empty lines between the imports. Sorry the tool has a strong opinion about standard v.s. third-party imports-- imports in different categories need to be separated by empty lines. 
```
diff --git a/examples/deployment_crud.py b/examples/deployment_crud.py
index 4fd02a7..7dcd6b3 100644
--- a/examples/deployment_crud.py
+++ b/examples/deployment_crud.py
@@ -17,7 +17,9 @@ Creates, updates, and deletes a deployment using AppsV1Api.
 """
 
 import datetime
+
 import pytz
+
 from kubernetes import client, config
 
 DEPLOYMENT_NAME = "nginx-deployment"
ERROR: InvocationError for command /home/travis/build/kubernetes-client/python/scripts/update-pycodestyle.sh (exited with code 1)
```

Reply:
@roycaihw, @yliaog, finally everything is green now :smile: 

Sorry, it caused a lot of trouble due to repeated failing tests (& me not understanding what exactly it required). But, I definitely learnt a lot about the tool.

Thank you so much.